website/docs/docs/contributing/building-a-new-adapter.md

```diff
@@ -266,7 +264,7 @@ To define an "adapter macro", use the `adapter_macro` function as shown [here](h
 {% macro create_table_as(temporary, relation, sql) -%}
 
   {# dbt will dispatch the macro call to the relevant macro #}
-  {{ adapter_macro('create_table_as', temporary, relation, sql) }}
+  {{ adapter.dispatch('create_table_as')(temporary, relation, sql) }}
```

Comment:
In dbt-utils, we wrap these macros in a `return` function:
```sql
https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/cross_db_utils/dateadd.sql
```

([Source](https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/cross_db_utils/dateadd.sql)

Is utils wrong, is this line wrong, or are they achieving different things? 🤔

Reply:
Good catch. `return()` is always a good idea, in case the macro result type is not a string. If there isn't a `return()`, whatever code the macro has templated out (including whitespace) is coerced to string and returned. For a macro like `dateadd`, which returns a string, it's not strictly necessary—should work either way—though the resulting whitespace may differ.

I'll add a `return()` here for consistency!